src/web100srv.c

```diff
@@ -1315,15 +1110,66 @@ int run_test(tcp_stat_agent *agent, Connection *ctl, TestOptions *testopt,
              vars.Timeouts, vars.Sndbuf, vars.MaxRwinRcvd, vars.CurrentCwnd,
              mismatch, bad_cable, totalcnt, refresh);
   }
-  if (ctl->ssl != NULL) {
-    SSL_shutdown(ctl->ssl);
-  } else {
-    shutdown(ctl->socket, SHUT_WR);
-  }
-  /* shutdown(ctlsockfd, SHUT_RDWR); */
+  close_connection(ctl);
+  shutdown(ctl->socket, SHUT_WR);
   return (0);
 }
 
+/** Read a single message sent from the parent server process to the child
+ * process, and perform and necessary communication that message necessitates.
+ * @param parent_pipe where to read the message from
+ * @param ctl the Connection to use for the network communication
+ * @param testopt the parameters of the test
+ * @param t_opt the tests the client supports
+ * @return the message value on success, a negative error code on failure.
+ */
+int read_from_parent(int parent_pipe, Connection *ctl, TestOptions *testopt,
```

Comment:
All we want out of `testopt` is the `connection_flags` so can we just take that as a param? It would reduce the confusion between `testopt` (also the name of a global we're shadowing) and `t_opt`.

Similarly, all we want out of `t_opts` is whether `TEST_STATUS` is set, so can we just pass a `do_heartbeat_check` param?